README.md

```diff
@@ -706,6 +706,8 @@ R8 / ProGuard
 
 Moshi contains minimally required rules for its own internals to work without need for consumers to embed their own. However if you are using reflective serialization and R8 or ProGuard, you must add keep rules in your proguard configuration file for your reflectively serialized classes.
 
+**Note on enums:** any enum classes (java or kotlin) used with moshi must be annotated with `@JsonClass(generateAdapter = false)`. Not annotating the class will result in `NoSuchFieldException` being thrown at adapter creation stage.
```

Comment:
```suggestion
**Note on enums:** any enum classes (Java or Kotlin) used with Moshi must be annotated with `@JsonClass(generateAdapter = false)`. Not annotating the class can result in `NoSuchFieldException` at runtime if the member represented in JSON isn't present in the enum.
```

Reply:
@ZacSweers, thanks for your input! I'm OK accepting your suggestions - just want to make sure we're on the same page on the actual issue:
the issue I was describing in #1261 will always happen for any obfuscated enum before Moshi even gets to reading/writing JSON. The exception is thrown at adapter creation time in `EnumJsonAdapter` constructor. `@JsonClass(generateAdapter = false)` fixes this as proguard/R8 is instructed to leave enum fields untouched.

The changes you suggested imply a slightly different perspective - outcome depends on the JSON contents, which I believe is not accurate in this case.

Reply:
@ZacSweers, I wonder if you had a chance to look through my reply?

Reply:
It does matter and it's a little more detailed than your description. I'd rather overcommunicate the risks here

Reply:
So, 2 things:
1. I'm not sure how I feel about this interaction... I was asked to submit a PR upon opening a defect which I did. Now I'm getting vague feedback that is very easy to interpret like "just accept my change and let's be done with this". I don't want to look negative and I totally understand and accept that you may have other more important things going on in your day-to-day life. I guess I'd just like to see an actual discussion if I'm asked to be part of that discussion. The discussion itself can wait until a better time.
2. > I'd rather overcommunicate the risks here
Well, I don't think the message you suggested is overcommunicating. If I understand correctly, it covers cases where proguard renames enums, and those do not match what is in JSON anymore. But the thing is - any obfuscated enum present anywhere in the object tree fed to `Moshi` will cause the adapter creation (`moshi.adapter(...)`) to fail with `NoSuchFieldException`. 
To put this in context, the regular android application would be failing at `Retrofit` service method invocation time. There will be no network request executed, no JSON received. For that matter, contents of the received JSON are also irrelevant in this specific context... Please let me know if my understanding here is not correct.